src/components/DailyStatistics/DailyStatisticsScreen.tsx

```diff
@@ -0,0 +1,266 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useContext,
+  useCallback,
+  useState,
+  createContext
+} from "react";
+import { View, StyleSheet } from "react-native";
+import { size, fontSize } from "../../common/styles";
+import { TopBackground } from "../Layout/TopBackground";
+import { Credits } from "../Credits";
+import { useConfigContext } from "../../context/config";
+import { CampaignConfigContext } from "../../context/campaignConfig";
+import { withNavigationFocus } from "react-navigation";
+import { TitleStatistic } from "./TitleStatistic";
+import { Sentry } from "../../utils/errorTracking";
+import { HelpButton } from "../Layout/Buttons/HelpButton";
+import { HelpModalContext } from "../../context/help";
+import { FeatureToggler } from "../FeatureToggler/FeatureToggler";
+import { Banner } from "../Layout/Banner";
+import { ImportantMessageContentContext } from "../../context/importantMessage";
+import { KeyboardAvoidingScrollView } from "../Layout/KeyboardAvoidingScrollView";
+import { AuthContext } from "../../context/auth";
+import { getDailyStatistics } from "../../services/statistics";
+import { summariseTransactions } from "./utils";
+import { TransactionHistoryCard } from "./TransactionHistoryCard";
+import { StatisticsHeader } from "./StatisticsHeader";
+import { addDays, subDays, getTime } from "date-fns";
+import {
+  AlertModalContext,
+  systemAlertProps,
+  ERROR_MESSAGE
+} from "../../context/alert";
+import { navigateHome } from "../../common/navigation";
+import { NavigationProps } from "../../types";
+
+interface StatisticsContext {
+  totalCount: number | null;
+  currentTimestamp: number;
+  lastTransactionTime: number | null;
+  transactionHistory: { name: string; category: string; quantity: number }[];
+  setTotalCount: (totalCount: number | null) => void;
+  setCurrentTimestamp: (currentTimestamp: number) => void;
+  setLastTransactionTime: (lastTransactionTime: number) => void;
+  setTransactionHistory: (
+    transactionHistory: { name: string; category: string; quantity: number }[]
+  ) => void;
+  clearStatistics: () => void;
+}
+
+export const StatisticsContext = createContext<StatisticsContext>({
+  totalCount: null,
+  currentTimestamp: Date.now(),
+  lastTransactionTime: null,
+  transactionHistory: [],
+  setTotalCount: () => null,
+  setCurrentTimestamp: () => null,
+  setLastTransactionTime: () => null,
+  setTransactionHistory: () => null,
+  clearStatistics: () => null
+});
+
+export const StatisticsContextProvider: FunctionComponent = ({ children }) => {
```

Comment:
actually the idea to have it as a context was so that in the future when we move on to location/admin or analytics across different campaigns, it would be useful!

Reply:
But the context is getting cleared on click of "next" and "prev", how does that tie in with the location/admin/different campaign analytics?

Reply:
i was thinking a bit longer term, like we store total count as a context between multiple campaigns and stuff like that, thats why i kept it as is, just that we moved it down into `DailyStatisticsScreen` for now since its the only screen thats using it!

Reply:
You mean if the user is signed onto multiple campaigns and we want to display total collected over all the campaigns? Hmm but I don't understand what kind of data we will be storing in the context since we are only fetching per day, and we are clearing our that data every time we click on prev/next day

Unless you are sure we are going to be sharing data across components in the near future, my preference is to keep this as a hook, since context is meant for sharing data across components. I am concerned we are doing this too early and end up never use this context at all but have no time to clean up too :p But just my preference, will leave it to you.

Reply:
hmmmm okay! because i think there were future plans of having like monthly view, calendar style, across campaigns and all that. that would be helpful right?

Reply:
Copy pasted from Slack convo

>For context vs hook, what I meant was
What do you plan to store in the context? Usually we store data that is fetched once and used everywhere. But right now on click of prev/next day, the context data changes. So actually we write to the context for nothing.
In the future, if I go to statistics feature and click prev day for example. Then I go to monthly view. Is the prev day data not going to be refetched? Or we are just going to overwrite the context with another fetch anyway?
>^ if we are consistently just fetching and overwriting the data in the store, then what is the point of putting in store?

But oh well seems like we are pressured for delivery

Reply:
oic my bad, didn't see a reply so i thought the discussion was still open